python/lsst/meas/astrom/matchOptimisticB.py

```diff
@@ -318,14 +228,23 @@ def matchObjectsToSources(self, refCat, sourceCat, wcs, refFluxField, maxMatchDi
             numUsableSources=numUsableSources,
             minMatchedPairs=minMatchedPairs,
             maxMatchDist=maxMatchDist,
-            sourceInfo=sourceInfo,
+            sourceFluxField=self.sourceSelector.fluxField,
             verbose=debug.verbose,
         )
 
         # cull non-good sources
         matches = []
+        # This is a hard coded version of the isGood flag from the old SourceInfo class that used to be
+        # part of this class. This is done current as the API for sourceSelector does not currently
```

Comment:
Also added another function for loading the schema keys since it wouldn't make sense to put it in the loop to test the isGood flag.